hyperspy/_components/physical_background.py

```diff
@@ -0,0 +1,205 @@
+            for i in range (0,len(w)):
+                if "Ka" in w [i].metadata.General.title :
+                    u[:,:,i] =w[i].data
+                elif "La" in w [i].metadata.General.title:
+                    u[:,:,i] =w[i].data*2.5
+                elif "La" in w [i].metadata.General.title:
+                    u[:,:,i] =w[i].data*2.8
+                g=np.ones([np.shape(w[0])[0],np.shape(w[0])[1], len(model._signal.metadata.Sample.elements)] )
+                for i in range (0,len(model._signal.metadata.Sample.elements)):
+                    g[:,:,i] =(u[:,:,i]/(np.sum(u,axis=-1))) *100          	    
+
+        elif len(np.shape(w[0]))==1 and np.shape(w[0])[0]>1 : #for 2D Data
+            u=np.ones([np.shape(w[0])[0], len(model._signal.metadata.Sample.elements)] )
+            for i in range (0,len(w)):
+                if "Ka" in w [i].metadata.General.title :
+                    u[:,i] =w[i].data
+                elif "La" in w [i].metadata.General.title:
+                    u[:,i] =w[i].data*2.5
+                elif "La" in w [i].metadata.General.title:
+                    u[:,i] =w[i].data*2.8
+            g=np.ones([np.shape(w[0])[0], len(model._signal.metadata.Sample.elements)] )
+            for i in range (0,len(model._signal.metadata.Sample.elements)):
+                g[:,i] =(u[:,i]/(np.sum(u,axis=-1))) *100
+
+        elif len(np.shape(w[0]))>1 and np.shape(w[0])[1]==1 : #for 2D Data
+            u=np.ones([np.shape(w[0])[0],np.shape(w[0])[1], len(model._signal.metadata.Sample.elements)] )
+            for i in range (0,len(w)):
+                if "Ka" in w [i].metadata.General.title :
+                    u[:,:,i] =w[i].data
+                elif "La" in w [i].metadata.General.title:
+                    u[:,:,i] =w[i].data*2.5
+                elif "La" in w [i].metadata.General.title:
+                    u[:,:,i] =w[i].data*2.8
+                g=np.ones([np.shape(w[0])[0],np.shape(w[0])[1], len(model._signal.metadata.Sample.elements)] )
+                for i in range (0,len(model._signal.metadata.Sample.elements)):
+                    g[:,:,i] =(u[:,:,i]/(np.sum(u,axis=-1))) *100     
+        
+        else:
+            u=np.ones([len(model._signal.metadata.Sample.elements)] ) #for 1D 
+            for i in range (0,len(w)):
+                if "Ka" in w [i].metadata.General.title :
+                    u[i] =w[i].data
+                elif "La" in w [i].metadata.General.title:
+                    u[i] =w[i].data*2.5
+                elif "La" in w [i].metadata.General.title:
+                    u[i] =w[i].data*2.8        
+            g=np.ones([len(model._signal.metadata.Sample.elements)] )
+            t=u.sum() 
+            for i in range (0,len(u)):
+                g[i] =u[i] /t*100
+    elif type(quantification) is np.ndarray: # if quantification is already an array of weight percent, directly keep the array
```

Comment:
This condition doesn't work very well. It would be easier if list are also admited. The problem is that the result of s.quantification is also a list.